### PR TITLE
Support credBlob for non-resident credentials

### PR DESCRIPTION
**Pull request by hcyang-google**
- Add a upper limit of max_cred_blob_length
- Add test cases for cred_blob in non-resident flows
- Modify the test helper functions in ctap/mod.rs a bit

### CODE REVIEW

File: src/ctap/credential_id.rs [hunk 1]
```diff
@@ -47,6 +47,7 @@ struct CredentialSource {
     private_key: PrivateKey,
     rp_id_hash: [u8; 32],
     cred_protect_policy: Option<CredentialProtectionPolicy>,
+    cred_blob: Option<Vec<u8>>,
 }
 
 // The data fields contained in the credential ID are serizlied using CBOR maps.
```
Comment (kaczmarczyck) [inline src/ctap/credential_id.rs]: Just noticed I made a type here. Fix this for me please? :)
"serizlied"

File: src/ctap/credential_id.rs [hunk 2]
```diff
@@ -112,10 +116,16 @@ fn decrypt_cbor_credential_id(
             } else {
                 None
             };
+            let cred_blob = if let Some(blob) = cred_blob {
+                Some(extract_byte_string(blob)?)
+            } else {
+                None
+            };
```
Comment (kaczmarczyck) [inline src/ctap/credential_id.rs]: Here and with the above `cred_protect_policy`, a shorter way to write this is:
```
let cred_blob = cred_blob.map(extract_byte_string).transpose()?;
```

File: src/ctap/credential_id.rs [hunk 3]
```diff
@@ -162,12 +172,14 @@ pub fn encrypt_to_credential_id(
     private_key: &PrivateKey,
```
Comment (kaczmarczyck) [inline src/ctap/credential_id.rs:166]: The description needs to be updated:
`/// Encrypts the given private key, relying party ID hash, and cred protect policy into a credential ID.`

Maybe a more general phrasing is easier to maintain.

**Review by kaczmarczyck — COMMENTED** [review]
Few nits, otherwise LGTM.

File: src/ctap/credential_id.rs (hunk 1 above)
Comment (hcyang-google) [reply src/ctap/credential_id.rs]: done

File: src/ctap/credential_id.rs (hunk 3 above)
Comment (hcyang-google) [reply src/ctap/credential_id.rs:166]: done

File: src/ctap/credential_id.rs (hunk 2 above)
Comment (hcyang-google) [reply src/ctap/credential_id.rs]: done

**Review by kaczmarczyck — APPROVED** [review]

**Pull request merged**